src/components/TextInput.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { ReactElement } from 'react';
+import {
+    FormControl as FormControlMUI,
+    InputLabel as InputLabelMUI,
+    Input as InputMUI,
+    FilledInput as FilledInputMUI,
+    SxProps,
+    Theme,
+} from '@mui/material';
+
+interface TextInputProps {
+    /**
+     * HTML element id
+     */
+    id?: string;
+    /**
+     * HTML input element type, defaults to text
+     */
+    type?: 'text' | 'username' | 'email' | 'password' | 'search' | 'url';
+    /**
+     * HTML element name
+     */
+    name?: string;
+    /**
+     * Label to display in the input field
+     */
+    label?: string;
```

Comment:
For accessibility, I think we might just want to make this prop required.

Reply:
Are you referring to the label prop? Would the comment generally refer to the line of code right above it, in this case line 27?

Reply:
Yep that is correct, It generally is the line of code right above, sometime it will also include the lines above if you select them. In this case it is just line 27.